### PR TITLE
Added validate Mac address functions to utility.

### PR DESCRIPTION
**Pull request by manvanthar**
Added validate Mac address functions to utility.

With misc tests changing from interface based to MAC address need this function to validate the MAC is valid.

Signed-off-by: Manvanthara B Puttashankar <manvanth@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by manvanthar** [comment]
Valid MAC address:
            36:84:37:5a:ea:02

        Invalid MAC address:
            3684375aea02
            36:84:37-5a-ea-02
            36-84-37-5a-ea-02
            36.84.37.5a.ea

**Review by PraveenPenguin — COMMENTED** [review]
@manvanthar thanks for patch ,can you please address the style  issue which is explained here 

https://github.com/avocado-framework/avocado/actions/runs/4124346912/jobs/7123826023#step:8:56

File: avocado/utils/network/interfaces.py [hunk 1]
```diff
@@ -781,3 +782,27 @@ def validate_ipv4_netmask_format(self, netmask):
...
+        This method checks if the mac address is 12 digit hexadecimal number.
+        Valid:
+            36:84:37:5a:ea:02
+
+        Invalid:
+            3684375aea02
+            36:84:37-5a-ea-02
+            36-84-37-5a-ea-02
+            36.84.37.5a.ea
+
```
[the remark this answers is not in the document]
Comment (FarooqAbdulla01) [reply avocado/utils/network/interfaces.py:797]: @manvanthar validation fails for "36:84:37:5a:ea:0ff", will have limit to 2 digits after colon with regex match.

File: avocado/utils/network/interfaces.py [hunk 2]
```diff
@@ -781,3 +782,35 @@ def validate_ipv4_netmask_format(self, netmask):
                 return False
             first_bit = False
         return True
+
+    def validate_macaddr(self, mac_str):
```
Comment (abdhaleegit) [inline avocado/utils/network/interfaces.py:786]: please change it to validate_mac_addr just for readability., rest all looks good

**Review by abdhaleegit — CHANGES_REQUESTED** [review]
just slight change in function name, also the parameter name can be mac_id instead mac_str

**Comment by FarooqAbdulla01** [comment]
validation failing for "3684375aea02" MAC address format.

following is the mac address set am validating against it.

validate_macaddr("36:84:37-5a-ea-02")
validate_macaddr("36-84-37-5a-ea-02")
validate_macaddr("3684375aea02")   ---> failing for this format, rest looks good.
validate_macaddr("36.84.37.5a.ea")
validate_macaddr("36:84:37:5a:ea:0ff")

**Comment by manvanthar** [comment]
> validation failing for "3684375aea02" MAC address format.
> 
> following is the mac address set am validating against it.
> 
> validate_macaddr("36:84:37-5a-ea-02") validate_macaddr("36-84-37-5a-ea-02") validate_macaddr("3684375aea02") ---> failing for this format, rest looks good. validate_macaddr("36.84.37.5a.ea") validate_macaddr("36:84:37:5a:ea:0ff")

It is updated in the comment on valid v/s invalid acceptable values.

**Comment by abdhaleegit** [comment]
@manvanthar I was trying to test the, please move this network/host.py the reason being the interface.py gets instantiated only when after a valid mac mapped to interface. so in our tests we have to exit way before this. so please move 
sorry for last minute change

**Comment by manvanthar** [comment]
> @manvanthar I tested the code, please move this to network/host.py (after get_interface_by_hwaddr) the reason being the interface.py gets instantiated only when after a valid mac mapped to interface. so in our tests we have to exit way before this. so please move
> sorry for last minute change

Sure Abdul. Have sent the updated code!

**Review by PraveenPenguin — APPROVED** [review]
LGTM

**Review by abdhaleegit — APPROVED** [review]
LGTM, tested with latest changes

```
JOB ID     : d4e4ebcac5e554f8faf753c6b778df551c3e7659
JOB LOG    : /home/OpTest/avocado-fvt-wrapper/results/job-2023-02-13T07.49-d4e4ebc/job.log
 (01/15) ethtool_test.py:Ethtool.test_ethtool;run-args-PrivateFlag-18ea: STARTED
 (01/15) ethtool_test.py:Ethtool.test_ethtool;run-args-PrivateFlag-18ea: PASS (14.55 s)
 (02/15) ethtool_test.py:Ethtool.test_ethtool;run-args-Driver-b3f8: STARTED
 (02/15) ethtool_test.py:Ethtool.test_ethtool;run-args-Driver-b3f8: PASS (8.34 s)
 (03/15) ethtool_test.py:Ethtool.test_ethtool;run-args-Autonegotiation-f53d: STARTED
 (03/15) ethtool_test.py:Ethtool.test_ethtool;run-args-Autonegotiation-f53d: PASS (22.98 s)
 (04/15) ethtool_test.py:Ethtool.test_ethtool;run-args-PauseParameter-3cac: STARTED
 (04/15) ethtool_test.py:Ethtool.test_ethtool;run-args-PauseParameter-3cac: PASS (12.35 s)
 (05/15) ethtool_test.py:Ethtool.test_ethtool;run-args-Coalescing-20f7: STARTED
 (05/15) ethtool_test.py:Ethtool.test_ethtool;run-args-Coalescing-20f7: PASS (12.45 s)
 (06/15) ethtool_test.py:Ethtool.test_ethtool;run-args-RingParameter-a769: STARTED
 (06/15) ethtool_test.py:Ethtool.test_ethtool;run-args-RingParameter-a769: PASS (12.32 s)
 (07/15) ethtool_test.py:Ethtool.test_ethtool;run-args-ShowOffloadProtocolFeature-b20b: STARTED
 (07/15) ethtool_test.py:Ethtool.test_ethtool;run-args-ShowOffloadProtocolFeature-b20b: PASS (13.27 s)
 (08/15) ethtool_test.py:Ethtool.test_ethtool;run-args-AdapterInitiate-26f1: STARTED
 (08/15) ethtool_test.py:Ethtool.test_ethtool;run-args-AdapterInitiate-26f1: PASS (19.60 s)
 (09/15) ethtool_test.py:Ethtool.test_ethtool;run-args-Statistics-3174: STARTED
 (09/15) ethtool_test.py:Ethtool.test_ethtool;run-args-Statistics-3174: PASS (12.40 s)
 (10/15) ethtool_test.py:Ethtool.test_ethtool;run-args-SelfTest-4acd: STARTED
 (10/15) ethtool_test.py:Ethtool.test_ethtool;run-args-SelfTest-4acd: PASS (22.11 s)
 (11/15) ethtool_test.py:Ethtool.test_ethtool;run-args-Show-permaddr-b691: STARTED
 (11/15) ethtool_test.py:Ethtool.test_ethtool;run-args-Show-permaddr-b691: PASS (12.40 s)
 (12/15) ethtool_test.py:Ethtool.test_ethtool;run-args-Show-time-stamping-ca91: STARTED
 (12/15) ethtool_test.py:Ethtool.test_ethtool;run-args-Show-time-stamping-ca91: PASS (12.32 s)
 (13/15) ethtool_test.py:Ethtool.test_ethtool;run-args-Get-phy-tunable-7b4c: STARTED
 (13/15) ethtool_test.py:Ethtool.test_ethtool;run-args-Get-phy-tunable-7b4c: WARN: Test passed but there were warnings during execution. Check the log for details. (12.63 s)
 (14/15) ethtool_test.py:Ethtool.test_ethtool;run-args-Show-channels-cba0: STARTED
```

**Pull request merged**